walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -15,9 +15,172 @@
 package com.google.sps;
 
 import java.util.Collection;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.lang.Math;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.HashMap;
+import java.util.Map;
 
 public final class FindMeetingQuery {
+
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
+    
+    // Edge case if request's duration is over a day -> returns no options
+    if (request.getDuration() > TimeRange.WHOLE_DAY.duration()){
+        return Arrays.asList();
+    }
+    // Get all mandatory meeting attendees
+    Set<String> attendees = new HashSet<>(request.getAttendees()); 
+
+    // Get all optional attendees
+    Set<String> optAttendees = new HashSet<>(request.getOptionalAttendees());
+    
+    // Process which events are from mandatory or optional attendees
+    ArrayList<TimeRange> mandatoryEvents = new ArrayList<>();
+    ArrayList<TimeRange> optEvents =  new ArrayList<>();
```

Comment:
Be consistent in naming; if you're going to write `mandatory`, write `optional` out as well.